api/submission/fetch_submission_history.go

```diff
@@ -0,0 +1,49 @@
+package submission
+
+import (
+    "github.com/eriq-augustine/autograder/api/core"
+    "github.com/eriq-augustine/autograder/db"
+    "github.com/eriq-augustine/autograder/model"
+)
+
+type FetchSubmissionHistoryRequest struct {
+    core.APIRequestAssignmentContext
+    core.MinRoleGrader
+
+    TargetUser       core.TargetUserSelfOrGrader `json:"target-email"`
+}
+
+type FetchSubmissionHistoryResponse struct {
+    FoundUser        bool `json:"found-user"`
+    FoundSubmissions bool `json:"found-submissions"`
+    GradingResults []*model.GradingResult `json:"grading-results"`
+}
+
+func HandleFetchSubmissionHistory(request *FetchSubmissionHistoryRequest) (*FetchSubmissionHistoryResponse, *core.APIError) {
+    response := FetchSubmissionHistoryResponse{
+        FoundUser:        false,
+        FoundSubmissions: false,
+        GradingResults:   make([]*model.GradingResult, 0),
+    }
+
+    if !request.TargetUser.Found {
+        return &response, nil;
+    }
+
+    response.FoundUser = true;
+
+    gradingResults, err := db.GetSubmissionResultHistory(request.Assignment, request.TargetUser.Email);
+    if err != nil {
+        return nil, core.NewInternalError("-604", &request.APIRequestCourseUserContext, "Failed to get submissions.").
+            Err(err).Add("email", request.TargetUser.Email);
+    }
+
+    if gradingResults == nil {
+        return &response, nil;
+    }
+
+    response.FoundSubmissions = true;
+    response.GradingResults = gradingResults;
+
+    return &response, nil;
+}
```

Comment:
Looks like some files are missing newlines at the end.
I would check your editor settings.

Reply:
Committing a fix for this.